### PR TITLE
Editor refactor fixup

### PR DESCRIPTION
**Pull request by facchinm**
Fixes some of the bugs emerged with recent Editor refactor.
@matthijskooijman @cmaglie take a look if you have some spare minutes (most fixes are trivial)
@per1234 could you test the build that @ArduinoBot will produce to confirm that #5402 and #5415 are gone? Thanks

### CODE REVIEW

File: arduino-core/src/processing/app/BaseNoGui.java [hunk 1]
```diff
@@ -162,12 +162,14 @@ static public PreferencesMap getBoardPreferences() {
...
     // Add all tools dependencies from the (possibily) referenced core
     String core = prefs.get("build.core");
-    if (core.contains(":")) {
+    if (core != null && core.contains(":")) {
       String split[] = core.split(":");
       TargetPlatform referenced = BaseNoGui.getCurrentTargetPlatformFromPackage(split[0]);
-      ContributedPlatform referencedPlatform = indexer.getContributedPlaform(referenced);
-      if (referencedPlatform != null)
-        requiredTools.addAll(referencedPlatform.getResolvedTools());
+      if (referenced != null) {
```
Comment (matthijskooijman) [inline arduino-core/src/processing/app/BaseNoGui.java]: Shouldn't an error be given when a referenced core does not exist?

**Comment by matthijskooijman** [comment]
@facchinm, I looked over the code and added some comments. Other than that, the commits look good to me. Thanks for diving into these!

**Comment by facchinm** [comment]
The tab reordering code is not optimal; sketch list and tabs list are completely independent, but the first one gets ordered every time we add a tab, while the second one does not. 
The `reorderTabs` function is there to avoid polluting the code with many calls for all situations; it actually executes something only if the sketch has changed, so the overhead is low (and in conjunction with 4e84238 it helps solving https://github.com/arduino/Arduino/issues/5492).
However, if there was a way to avoid duplicating the comparator code it would be great :slightly_smiling_face:

**Comment by matthijskooijman** [comment]
If the list of sketch files is already kept sorted, can't we use the position of a sketch file in the sketch file list as the position when creating a new tab? Or, whenever the list of sketch files changes, re-create the list of tabs or something?

**Comment by per1234** [comment]
>@per1234 could you test the build that @ArduinoBot will produce to confirm that #5402 and #5415 are gone?

@facchinm, I confirm those issues are fixed with the test build. Thanks!

**Comment by cmaglie** [comment]
* added the error message when the referenced core is missing
* solved the code duplication problem with a lambda
* fixed a small bug (new tab is auto-selected after creation, before it was selected always the last one)

**Comment by matthijskooijman** [comment]
A bit late, but the extra commits look good to me :-)

**Pull request merged**